testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/providers/jackson2/whitelist/WhiteListPolymorphicTypeValidatorTest.java

```diff
@@ -0,0 +1,150 @@
+package org.jboss.resteasy.test.providers.jackson2.whitelist;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.logging.Logger;
+import org.jboss.resteasy.client.jaxrs.ResteasyClient;
+import org.jboss.resteasy.plugins.providers.jackson.WhiteListPolymorphicTypeValidatorBuilder;
+import org.jboss.resteasy.spi.HttpResponseCodes;
+import org.jboss.resteasy.test.providers.jackson2.whitelist.model.AbstractVehicle;
+import org.jboss.resteasy.test.providers.jackson2.whitelist.model.TestPolymorphicType;
+import org.jboss.resteasy.test.providers.jackson2.whitelist.model.air.Aircraft;
+import org.jboss.resteasy.test.providers.jackson2.whitelist.model.land.Automobile;
+import org.jboss.resteasy.utils.PortProviderUtil;
+import org.jboss.resteasy.utils.TestUtil;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import javax.ws.rs.client.ClientBuilder;
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+
+/**
+ * @tpSubChapter Jackson2 provider
+ * @tpChapter Integration tests
+ * @tpSince RESTEasy 4.5.0
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class WhiteListPolymorphicTypeValidatorTest {
+
+    protected static final Logger logger = Logger.getLogger(WhiteListPolymorphicTypeValidatorTest.class.getName());
+
+    static ResteasyClient client;
+
+    @Deployment(name = "default")
+    public static Archive<?> deploy() {
+        WebArchive war = TestUtil.prepareArchive(WhiteListPolymorphicTypeValidatorTest.class.getSimpleName());
+        war.addClass(WhiteListPolymorphicTypeValidatorTest.class);
+        Map<String, String> contextParam = new HashMap<>();
+        contextParam.put("jackson.deserialization.whitelist.allowIfBaseType.prefix", Automobile.class.getPackage().getName());
+        contextParam.put("jackson.deserialization.whitelist.allowIfSubType.prefix", Automobile.class.getPackage().getName());
```

Comment:
Actually I believe that in this case the latter context param would be enough.

Reply:
Ok, I will add another test. Thanks for a review.

Reply:
I've tried to add a test enabled by a system property, it however broke the current `testGood` method. It seems the system property settings override context params, i.e. MP Config. If so, it would need a separate surefire execution to test both. See
https://github.com/resteasy/Resteasy/commit/742cb04ad91956e849b4307dede30b584f3df326